### ISSUE TITLE
Reference line overlapping the tooltip

### ISSUE THREAD

**Issue by agonselimi** [Abhinandan-Kushwaha/react-native-gifted-charts#476]
**Describe the bug**
I'm using BarChart component to display data on my app. I wanted to use a reference line as a total average, but it is overlapping my tooltip, and it doesn't look that good. 

**To Reproduce**
```
           <BarChart
            width={WINDOW_WIDTH - scale(60)}
            showReferenceLine1
            referenceLine1Config={{
              dashWidth: 3,
              dashGap: 3,
              type: "dashed",
              color: colors.white,
            }}
            referenceLine1Position={average}
            isAnimated
            animationDuration={300}
            noOfSections={4}
            stackData={stackData}
            rulesType="solid"
            rulesColor={colors.grey}
            xAxisColor={colors.grey}
            yAxisThickness={0}
            yAxisTextStyle={[styles.axisText, { color: colors.grey }]}
            xAxisLabelTextStyle={[styles.axisText, { color: colors.grey }]}
            barWidth={scale(12)}
            spacing={scale(7)}
            stepHeight={scale(40)}
            endSpacing={scale(250)}
            leftShiftForLastIndexTooltip={scale(20)}
            renderTooltip={(item) => (
              <View
                style={[
                  styles.toolTipCtn,
                  {
                    backgroundColor: colors.lighterBlack,
                    borderColor: colors.grey,
                  },
                ]}
              >
                <View style={[styles.row, styles.toolTip]} key="0">
                  <Text
                    style={[
                      styles.legendText,
                      { color: colors.grey, width: scale(120) },
                    ]}
                  >
                    Total Revenue
                  </Text>
                  <Text
                    style={[styles.legendText, { color: colors.primaryText }]}
                  >
                    € {item?.stacks?.[0]?.value + item?.stacks?.[1]?.value ?? 0}
                  </Text>
                </View>
                {item?.stacks?.map((el, index) => (
                  <View style={[styles.row, styles.toolTip]} key={index}>
                    <Text
                      style={[
                        styles.legendText,
                        { color: colors.grey, width: scale(120) },
                      ]}
                    >
                      {el?.label}
                    </Text>
                    <Text
                      style={[styles.legendText, { color: colors.primaryText }]}
                    >
                      € {el?.value}
                    </Text>
                  </View>
                ))}
              </View>
            )}
          />
          
          const styles = StyleSheet.create({
          toolTip: {
          justifyContent: "flex-start",
          },
          toolTipCtn: {
          padding: 10,
          borderRadius: 8,
          borderWidth: 1,
          position: "absolute",
          top: -scale(20),
          zIndex: 99999999999,
          },
          })
```

**Expected behavior**
The reference line should overlap the Bar chart columns, but not the tooltip.

**Screenshots**
<img width="380" alt="Screenshot 2023-12-21 at 10 25 39" src="https://github.com/Abhinandan-Kushwaha/react-native-gifted-charts/assets/52735859/f250ba98-6927-4ca3-b9c0-240eaca57794">

I'm using: 
"react-native": "0.72.5",
"expo": "~49.0.13",
"react-native-gifted-charts": "^1.3.11",

**Comment by Abhinandan-Kushwaha** [comment]
Hi @agonselimi 👋
I think this was fixed in version `1.3.25`
I would recommend you use the latest version of the library, that is `1.3.28`

You can take a look at https://github.com/Abhinandan-Kushwaha/react-native-gifted-charts/issues/310

**Comment by Abhinandan-Kushwaha** [comment]
Hey @agonselimi 
Try using the prop `referenceLinesOverChartContent={false}`. However, reference lines will then go even behind the bars.
It is available from version `1.3.29` onwards.

> The reference line should overlap the Bar chart columns, but not the tooltip.

This is currently not achievable.
